### PR TITLE
Fix SDL and Prefast Errors

### PR DESCRIPTION
**Pull request by nums11**
Fixes Errors 1978844, 1978870, 1978850, 1978855, and 9245

### CODE REVIEW

**Review by martinb35 — APPROVED** [review]

**Review by smk2007 — APPROVED** [review]

**Review by snnn — APPROVED** [review]
Thank you.

Please close the work items after this PR is merged.

**Comment by snnn** [comment]
All the tests passed. You may merge this PR if you want.

**Pull request merged**
